source/common/common/scope_tracker.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "envoy/common/scope_tracker.h"
+#include "envoy/event/dispatcher.h"
+
+namespace Envoy {
+
+class ScopeTrackerImpl {
```

Comment:
nit: I might call this `RAIIScopeTracker` or something like that? Since this isn't really an implementation of anything?

Reply:
I think this name is better future-proofed for when we use the scope object for customer memory/cpu accounting (the htuch@ feature) - WDYT?

Reply:
I don't feel strongly about it, other than to say that `Impl` usually is an implementation of an interface which this is not, so I would probably try to find a more descriptive name which IMO is some type of context or something along those lines. e.g. `ScopeTrackerScopeState` or something. Anyway, not a big deal either way.

Reply:
(I guess a small class comment on what it is for would be nice either way)